texar/data/data/multi_aligned_data.py

```diff
@@ -195,22 +191,23 @@ def __init__(self, hparams, device: Optional[torch.device] = None):
         self._embedding = self.make_embedding(self._hparams.datasets,
                                               self._vocab)
 
-        self._dataset_features: List[Optional[Dict[str, Any]]] = []
         self._name_prefix: List[str] = []
+        self._names: List[Dict[str, Any]] = []
         datasources: List[DataSource] = []
         filters: List[Optional[Callable[[str], bool]]] = []
-        for _, hparams_i in enumerate(self._hparams.datasets):
+        self._databases: List[DataBase] = []
+        for idx, hparams_i in enumerate(self._hparams.datasets):
             dtype = hparams_i.data_type
             datasource_i: DataSource
+
             if _is_text_data(dtype) or _is_scalar_data(dtype):
```

Comment:
I would prefer separating the logic of text data and scalar data, since the only thing they share is the construction of the `TextLineDataSource`. Separating them would make the code flow more clear.